File: nodejs/src/campaigns_example.js
const MailproClient = require('./mailproClient');

(async () => {
  try {
    const client = new MailproClient('YOUR_API_KEY');

    // Create a new campaign
    const campaignData = {
      name: 'My First Campaign',
      subject: 'Welcome!',
      from: 'you@example.com',
      html: '<h1>Hello Subscribers</h1><p>This is my first campaign!</p>'
    };
    const createRes = await client.createCampaign(campaignData);
    console.log('Campaign created:', createRes);

    // Fetch details of the campaign
    if (createRes.id) {
      const details = await client.getCampaign(createRes.id);
      console.log('Campaign details:', details);
    }
  } catch (err) {
    console.error('Error with campaigns:', err.message);
  }
})();

File: python/campaigns_example.py
const MailproClient = require('./mailproClient');

(async () => {
  const token = process.env.MAILPRO_TOKEN;
  const client = new MailproClient({ token });
  try {
    const campaignPayload = {
      name: 'SDK Starter Campaign',
      subject: 'Hello from SDK',
      idFrom: 12345,
      listIds: [1],
      html: '<h1>Campaign</h1><p>Example</p>'
    };
    const created = await client.createCampaign(campaignPayload);
    console.log('Created campaign:', created);

    if (created && created.id) {
      const status = await client.getCampaign(created.id);
      console.log('Campaign status:', status);

      // To send
      // const sendResp = await client.sendCampaign(created.id);
      // console.log('Send response:', sendResp);
    }
  } catch (err) {
    console.error('Campaign error:', err.message);
  }
})();
